### PR TITLE
Checks for empty citekeys and whitespaces.

### LINKED ISSUE #163: Records with no ID parsed incorrectly
Given the following record:

> @misc{
author = {AMAP},
title = {Summary – The Greenland Ice Sheet in a Changing Climate: Snow, Water, Ice and Permafrost in the Arctic (SWIPA)},
publisher = {Arctic Monitoring and Assessment Programme (AMAP)},
pages = {22 pp.},
year = {2009},
type = {Generic},
}

The parser sets ID as "author = {AMAP" and doesn't recognize the author field.

Changing `re.split(',\s*\n|\n\s*,', record)` to `re.split('(?<={)\s*\n|,\s*\n|\n\s*,', record)` line 239 of bparser.py seems to fix the problem.

### PR DESCRIPTION
**Pull request by omangin**
Fixes #163.

Prevents empty citekeys and whitespaces in citekeys at parse time.

### CODE REVIEW

File: bibtexparser/bibtexexpression.py [hunk 1]
```diff
@@ -149,9 +149,22 @@ def __init__(self):
         entry_type.setParseAction(first_token)
 
         # Entry key: any character up to a ',' without leading and trailing
-        # spaces.
-        key = pp.SkipTo(',')('Key')  # Exclude @',\#}{~%
-        key.setParseAction(lambda s, l, t: first_token(s, l, t).strip())
+        # spaces. Also exclude spaces and prevent it from being empty.
+        key = pp.SkipTo(',')('Key')  # TODO Maybe alsi exclude @',\#}{~%
+
+        def citekeyParseAction(s, l, t):
```
Comment (sciunto) [inline bibtexparser/bibtexexpression.py]: Please, add a short docstring to specify what do s, l and t correspond to. I'm not sure it's straightforward for everyone, even for the comment above.

File: bibtexparser/bibtexexpression.py [hunk 2]
```diff
@@ -149,9 +149,22 @@ def __init__(self):
         entry_type.setParseAction(first_token)
 
         # Entry key: any character up to a ',' without leading and trailing
-        # spaces.
-        key = pp.SkipTo(',')('Key')  # Exclude @',\#}{~%
-        key.setParseAction(lambda s, l, t: first_token(s, l, t).strip())
+        # spaces. Also exclude spaces and prevent it from being empty.
+        key = pp.SkipTo(',')('Key')  # TODO Maybe alsi exclude @',\#}{~%
```
Comment (sciunto) [inline bibtexparser/bibtexexpression.py]: s/alsi/also/

File: bibtexparser/bibtexexpression.py (hunk 1 above)
Comment (omangin) [reply bibtexparser/bibtexexpression.py]: `s, l, t` is the standard signature of `pyparsing` parse actions. It is indeed not clear so I will change to `string_, location, token`.

**Comment by omangin** [comment]
@sciunto Thanks for the feedback. Please merge if it is fine for you and I don't answer right away.

**Comment by sciunto** [comment]
Perfect! thanks @omangin

**Pull request merged**